### ISSUE TITLE
cpp:function can't parse multiple signatures

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#677]
The python domain allows multiple signatures to be grouped as follows:

```
.. py:function:: start(x)
                 end(x)
```

The corresponding formulation in the C++ domain does not parse:

```
.. cpp:function:: iterator begin()
                  iterator end()
```

In the general case, doing this correctly for C++ might be undecidable, due to cases like this:

```
.. cpp:function:: T& vector<T>::operator[](size_t loc)
                  const T& vector<T>::operator[](size_t loc) const
```

where it's ambiguous whether the `const` should be interpreted as a trailing part of the first signature, or the leading part of the second signature. However, it wouldn't be so bad to require an explicit comma between declarations.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/677
- Originally reported by: Anonymous
- Originally created at: 2011-04-17T07:09:58.444

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-22 08:40:50+00:00_

Fix #677: allow multiple signatures in C++ domain directives.

→ <<cset b3108306cda69378bbaca8e2ee968644d45d264e>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:32:19+00:00_

Removing version: 1.0.7 (automated comment)
